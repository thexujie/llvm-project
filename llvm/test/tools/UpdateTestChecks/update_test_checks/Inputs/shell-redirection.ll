;; Check that we can parse RUN lines that have shell redirections
; RUN: opt %s -passes=instsimplify -S -o - 2>/dev/null | FileCheck %s --check-prefix=I32
; RUN: opt < %s -passes=instsimplify -S 2>&1 | FileCheck %s --check-prefix=I32
; RUN: sed 's/i32/i64/g' %s | opt -passes=instsimplify -S | FileCheck %s --check-prefix=I64

define i32 @common_sub_operand(i32 %X, i32 %Y) {
  %Z = sub i32 %X, %Y
  %Q = add i32 %Z, %Y
  ret i32 %Q
}
