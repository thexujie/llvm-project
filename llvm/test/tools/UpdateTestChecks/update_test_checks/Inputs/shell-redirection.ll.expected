; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
;; Check that we can parse RUN lines that have shell redirections
; RUN: opt %s -passes=instsimplify -S -o - 2>/dev/null | FileCheck %s --check-prefix=I32
; RUN: opt < %s -passes=instsimplify -S 2>&1 | FileCheck %s --check-prefix=I32
; RUN: sed 's/i32/i64/g' %s | opt -passes=instsimplify -S | FileCheck %s --check-prefix=I64

define i32 @common_sub_operand(i32 %X, i32 %Y) {
; I32-LABEL: @common_sub_operand(
; I32-NEXT:    ret i32 [[X:%.*]]
;
; I64-LABEL: @common_sub_operand(
; I64-NEXT:    ret i64 [[X:%.*]]
;
  %Z = sub i32 %X, %Y
  %Q = add i32 %Z, %Y
  ret i32 %Q
}
